libsolidity/codegen/CompilerContext.h

```diff
@@ -62,11 +62,13 @@ class CompilerContext
 public:
 	explicit CompilerContext(
 		langutil::EVMVersion _evmVersion,
+		std::optional<uint8_t> _eofVersion,
```

Comment:
Since we didn't add this for passing it down to ``m_asm``, since it's guaranteed to be ``std::nullopt`` for the legacy pipeline, we can do the same now, i.e.: this is legacy codegen, we can just directly pass ``std::nullopt`` down everywhere instead of passing this in here (since it will *always* be ``std::nullopt`` anyways) - this includes for Yul dialects constructed for utility functions and inline assembly during legacy codegen. So whereever this PR currently uses ``CompilerContext::eofVersion()`` we can instead directly use ``std::nullopt``.

Reply:
Agreed. An assert that we're not using EOF where we create `Compiler` would be enough - and we already have one (in `CompilerStack`), so we can leave this out of here.

Reply:
Oh shit. I was sure that it's a part of general compiling path but now noticed that it's a codegen in libsolidity not libyul. Fair point. Will change accordigly